benchmarks/suites/polybench/cholesky/cholesky.cu

```diff
@@ -18,7 +17,7 @@ __global__ void kernel0(pbsize_t n, idx_t j, real *A) {
 __global__ void kernel1(pbsize_t n, idx_t j, real *A) {
   idx_t i = blockDim.x * blockIdx.x + threadIdx.x;
 
-  if (j < n && i > j)
+  if (j < n && i > j && (i * n + j) < (n * n) && (j * n + j) < (n * n))
```

Comment:
```suggestion
  if (i < n && i > j)
```
Should be sufficient. Since `j < i` and with `i < n`, implies `j < n` and the others with fewer operations.  I think my mistake here was just that switchup. In your version `i`/`j` may still be larger than `n` and the subscript still be less than $n^2$

Can you test it?

Reply:
I have tested it. It is working as expected. Thank you.